sdks/java/io/parquet/src/main/java/org/apache/beam/sdk/io/parquet/ParquetIO.java

```diff
@@ -1210,6 +1221,13 @@ public void open(WritableByteChannel channel) throws IOException {
               .withWriteMode(OVERWRITE)
               .withConf(SerializableConfiguration.newConfiguration(getConfiguration()))
               .withRowGroupSize(getRowGroupSize());
+      if (getAvroDataModel() != null) {
+        try {
+          builder.withDataModel(buildModelObject(getAvroDataModel()));
+        } catch (ReflectiveOperationException e) {
+          LOG.warn("Couldn't set the model: " + e.getMessage());
```

Comment:
Don't we need to fail a pipeline here?

Reply:
My initial idea was to not modify the previous behaviour: in this case if the exception is raised the model will be the default one set by the `AvroParquetWriter` class, which is the `SpecificData` one

Reply:
Which could be a non-sense since if a user specify a model this is silently failing and setting a different model. So I think you are right this should cause the pipeline to fail

Reply:
Yes, agree. If initial user's intent was to specify a model and it's not correct for some reasons, we need to fail with a proper error message.

Reply:
So, please, fix this one.

Reply:
Just pushed the changes